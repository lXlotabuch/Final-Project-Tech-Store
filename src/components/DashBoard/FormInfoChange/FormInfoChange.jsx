import React from 'react';
import {
  Form, Input, Button
} from 'antd';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { updateCustomer } from '../../../store/customer/middleware';
import { setHideModal } from '../../../store/dashBoardModal/middleware';
import { selectCustomerInfo } from '../../../store/customer/reducer'
import { validName, validTelephone } from '../../../utils/constants'

const mapStateToProps = (state) => ({
  info: selectCustomerInfo(state)
})

export const FormInfoChangeComponent = ({setHideModal, updateCustomer}) => {
  const [form] = Form.useForm();

  const onFinish = (values) => {
    if (Object.values(values).every((elem) => elem === undefined)) {
      setHideModal();
      form.resetFields();
      return
    }
    updateCustomer(values, 'Your contact information has been changed');
    setHideModal();
    form.resetFields()
  }
  return (
    <Form name="nest" form={form} onFinish={onFinish}>
      <Form.Item
        name="firstName"
        label="First name"
        rules={[
          {
            pattern: validName,
            message: 'Allowed characters is a-z, а-я.'
          },
          {
            min: 2,
            max: 25,
            message: 'First name must be beetwen 2 and 25 characters.'
          },
        ]}
      >
        <Input />
      </Form.Item>
      <Form.Item
        style={{paddingBottom: '0px'}}
        name="lastName"
        label="Last name"
        rules={[
          {
            pattern: validName,
            message: 'Allowed characters is a-z, а-я.'
          },
          {
            min: 2,
            max: 25,
            message: 'Last Name must be beetwen 2 and 25 characters.'
          },
        ]}
      >
        <Input />
      </Form.Item>
      <Form.Item
        name="telephone"
        label="New Phone"
        rules={[
          {
            pattern: validTelephone,
            message: 'Phone number must start with "+380", allowed characters is 0-9.'
          },
          {
            min: 13,
            max: 13,
            message: 'Phone number must contain 12 numbers.'
          }
        ]}
      >
        <Input />
      </Form.Item>
      <Form.Item>
        <Button type="primary" htmlType="submit">
          Submit
        </Button>
      </Form.Item>
    </Form>
  );
}

const FormInfoChange = connect(
  mapStateToProps, {
    setHideModal,
    updateCustomer
  }
)(FormInfoChangeComponent)

FormInfoChangeComponent.propTypes = {
  setHideModal: PropTypes.func,
  updateCustomer: PropTypes.func,
}
FormInfoChangeComponent.defaultProps = {
  setHideModal: () => null,
  updateCustomer: () => null,
}
export default FormInfoChange;
