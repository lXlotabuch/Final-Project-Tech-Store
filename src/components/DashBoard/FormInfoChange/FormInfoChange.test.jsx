import React from 'react'
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import { machMedia } from '../../../mocks/matchMedia.mock';
import { FormInfoChangeComponent } from './FormInfoChange'
import { store } from '../../../store/index'

describe('FormInfoChangeComponent', () => {
  beforeEach(() => {
    machMedia()
  })
  test('render', () => {
    const {asFragment} = render(
      <Provider store={store}>
        <FormInfoChangeComponent
          setHideModal={() => {}}
          updateCustomer={() => {}}
        />
      </Provider>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})