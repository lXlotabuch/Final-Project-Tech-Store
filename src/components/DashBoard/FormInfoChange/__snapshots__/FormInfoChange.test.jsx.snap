// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInfoChangeComponent render 1`] = `
<DocumentFragment>
  <form
    class="ant-form ant-form-horizontal"
    id="nest"
  >
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-label"
      >
        <label
          class=""
          for="nest_firstName"
          title="First name"
        >
          First name
        </label>
      </div>
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input"
              id="nest_firstName"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
      style="padding-bottom: 0px;"
    >
      <div
        class="ant-col ant-form-item-label"
      >
        <label
          class=""
          for="nest_lastName"
          title="Last name"
        >
          Last name
        </label>
      </div>
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input"
              id="nest_lastName"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-label"
      >
        <label
          class=""
          for="nest_telephone"
          title="New Phone"
        >
          New Phone
        </label>
      </div>
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input"
              id="nest_telephone"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <button
              class="ant-btn ant-btn-primary"
              type="submit"
            >
              <span>
                Submit
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
